File: Task_2/task_2.rb
require 'set'

def is_pangram
  print "Введите строку: "
  sentence = gets.chomp.downcase

  letters = Set.new
  sentence.each_char do |char|
    if char >= 'a' && char <= 'z'
        letters.add(char)
    end
  end
  letters.size == 26
end

if is_pangram
  puts "true"
else
  puts "false"
end

File: Task_2/task_2.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <cctype>

using namespace std;

bool isPangram() {
    cout << "Введите строку для проверки: ";
    string sentence;
    getline(cin, sentence);

    unordered_set<char> letters;
    for (char c : sentence) {
        char lowerC = tolower(c);
        if (lowerC >= 'a' && lowerC <= 'z') {
            letters.insert(lowerC);
        }
    }
    return letters.size() == 26;
}

int main() {
    if (isPangram()) {
        cout << "true" << endl;
    } else {
        cout << "false" << endl;
    }
    return 0;
}
